src/lib/routers/history.js

```diff
@@ -44,6 +50,13 @@ class BrowserHistory {
     this.writeDelay = writeDelay;
     this._createURL = createURL;
     this.parseURL = parseURL;
+
+    const routeState = this.parseURL({
+      qsModule: qs,
+      location: window.location,
+    });
+    const title = this.windowTitle && this.windowTitle(routeState);
```

Comment:
You can inline the call `this.read()` inside `this.windowTitle` (or make a condition) to avoid a useless parse when `windowTitle` is not provided.